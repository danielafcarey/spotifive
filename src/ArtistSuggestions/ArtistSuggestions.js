import React from 'react';

const ArtistSuggestions = ({ suggestions }) => {
  const optionList = suggestions.map((suggestion, i) => {
    return <option key={ i }>{ suggestion }</option>
  }).slice(0, 10);
  
  return (
    <datalist id="artist-suggestions">
      { optionList }
    </datalist>
  );
};

export default ArtistSuggestions;
