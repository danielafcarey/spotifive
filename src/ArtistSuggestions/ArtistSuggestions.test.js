import React from 'react';
import { shallow } from 'enzyme';
import ArtistSuggestions from './ArtistSuggestions';

describe('ArtistSuggestions', () => {

  it('matches the snapshot', () => {
    const mockProps = {
      suggestions: ['bob', 'stacy']
    }
    const wrapper = shallow(<ArtistSuggestions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })
})
