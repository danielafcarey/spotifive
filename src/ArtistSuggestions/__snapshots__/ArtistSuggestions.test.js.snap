// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSuggestions matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistSuggestions
    suggestions={
        Array [
            "bob",
            "stacy",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
          bob
</option>,
        <option>
          stacy
</option>,
      ],
      "id": "artist-suggestions",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "artist0",
        "nodeType": "host",
        "props": Object {
          "children": "bob",
        },
        "ref": null,
        "rendered": "bob",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "artist1",
        "nodeType": "host",
        "props": Object {
          "children": "stacy",
        },
        "ref": null,
        "rendered": "stacy",
        "type": "option",
      },
    ],
    "type": "datalist",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            bob
</option>,
          <option>
            stacy
</option>,
        ],
        "id": "artist-suggestions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "artist0",
          "nodeType": "host",
          "props": Object {
            "children": "bob",
          },
          "ref": null,
          "rendered": "bob",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "artist1",
          "nodeType": "host",
          "props": Object {
            "children": "stacy",
          },
          "ref": null,
          "rendered": "stacy",
          "type": "option",
        },
      ],
      "type": "datalist",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
